packages/react/src/render.tsx

```diff
@@ -0,0 +1,71 @@
+import { HTMLWidget, publish, SVGWidget } from "@hpcc-js/common";
+import * as React from "@hpcc-js/preact-shim";
+
+export function render<P>(C: React.FunctionComponent<P>, props: Readonly<P>, parent: Element | Document | ShadowRoot | DocumentFragment, replaceNode?: Element | Text) {
+    React.render(React.h(C, props), parent, replaceNode);
+}
+
+export function svgRender<P>(C: React.FunctionComponent<P>, props: Readonly<P>, parent: Element | Document | ShadowRoot | DocumentFragment, replaceNode?: Element | Text) {
+    React.render(React.h("svg", null, React.h(C, props)), parent, replaceNode);
+}
+
+export class HTMLAdapter<P> extends HTMLWidget {
+
+    @publish({}, "object", "Properties")
+    protected _props: P = {} as P;
+    props(): P;
+    props(_: Partial<P>): this;
+    props(_?: Partial<P>): P | this {
+        if (!arguments.length) return this._props;
+        this._props = { ...this._props, ..._ };
+        return this;
+    }
+
+    prop<K extends keyof P>(_: K): P[K];
+    prop<K extends keyof P>(_: K, value: P[K]): this;
+    prop<K extends keyof P>(_: K, value?: P[K]): this | P[K] {
```

Comment:
Will typescript recognize the first prop string argument and suggest valid values for the second?

Reply:
Nearly it will spot the first prop string and suggest `types` for the value (i.e. string, number, boolean etc.)